tools/testing/tools_test.go

```diff
@@ -0,0 +1,67 @@
+package toolstesting
+
+import (
+	"errors"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/symflower/eval-dev-quality/tools"
+)
+
+func TestRequiresTool(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		SetupMock    func(t *testing.T, m *MockTool)
+		ExpectFailed bool
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			tt := &testing.T{}
+			mock := NewMockToolNamed(t, "some-tool")
+			tc.SetupMock(t, mock)
+
+			// The "testing" logic internally calls "runtime.Goexit" in case of a failure which exits a Goroutine (but still runs "defer"s) so start a new one to not kill ourselves.
```

Comment:
`// The "testing" package calls "runtime.Goexit" in case of a failure which exits the innermost Goroutine. We therefore call the function under test in a new Goroutine such that the outer one gets killed.

Reply:
obsolete